src/Sulu/Bundle/AdminBundle/Resources/js/components/ColorPicker/tests/ColorPicker.test.js

```diff
@@ -1,78 +1,101 @@
-    colorPicker.update();
-    expect(colorPicker.find('Input').prop('valid')).toEqual(false);
+    expect(onChange).toHaveBeenCalledWith(undefined);
 
-    colorPicker.find('Input').instance().props.onChange('#ccc', {target: {value: '#ccc'}});
-    colorPicker.find('Input').instance().props.onBlur();
-    colorPicker.update();
-    expect(colorPicker.find('Input').prop('valid')).toBe(true);
+    await userEvent.type(input, '#ccc');
+    fireEvent.blur(input);
+
+    expect(input).toBeValid();
 });
 
-test('ColorPicker should trigger callbacks correctly', () => {
+test('ColorPicker should trigger callbacks correctly', async() => {
     const onChange = jest.fn();
     const onBlur = jest.fn();
-    const colorPicker = mount(<ColorPicker onBlur={onBlur} onChange={onChange} value={null} />);
+    render(<ColorPicker onBlur={onBlur} onChange={onChange} value="#abc" />);
+
+    const input = screen.queryByDisplayValue('#abc');
 
     // provide invalid value
-    colorPicker.find('Input').instance().props.onChange('xxx', {target: {value: 'xxx'}});
-    colorPicker.find('Input').instance().props.onBlur();
-    colorPicker.update();
+    await userEvent.type(input, 'xxx');
+    fireEvent.blur(input);
     expect(onChange).toBeCalledWith(undefined);
     expect(onBlur).toBeCalled();
 
     // provide one more invalid value
-    colorPicker.find('Input').instance().props.onChange('abc', {target: {value: 'abc'}});
-    colorPicker.find('Input').instance().props.onBlur();
-    colorPicker.update();
+    await userEvent.type(input, 'abc');
+    fireEvent.blur(input);
     expect(onChange).toBeCalledWith(undefined);
     expect(onBlur).toBeCalled();
 
     // now add a valid value
-    colorPicker.find('Input').instance().props.onChange('#abcabc', {target: {value: '#abcabc'}});
-    colorPicker.find('Input').instance().props.onBlur();
-    colorPicker.update();
-    expect(onChange).toBeCalledWith('#abcabc');
+    await userEvent.type(input, '#abc');
+    fireEvent.blur(input);
+    expect(onChange).toBeCalledWith('#abc');
     expect(onBlur).toBeCalled();
 
     expect(onBlur).toHaveBeenCalledTimes(3);
 });
 
-test('ColorPicker should call the correct callbacks when value from overlay was selected', () => {
+test('ColorPicker should call the correct callbacks when value from overlay was selected', async() => {
     const onChange = jest.fn();
     const onBlur = jest.fn();
-    const colorPicker = mount(<ColorPicker onBlur={onBlur} onChange={onChange} value={null} />);
+    const {baseElement} = render(<ColorPicker onBlur={onBlur} onChange={onChange} value="#abc" />);
+
+    const icon = screen.queryByLabelText('su-square');
+    await userEvent.click(icon);
+
+    expect(baseElement).toMatchSnapshot();
 
-    colorPicker.find('Icon').simulate('click');
-    colorPicker.find('ColorPicker').at(1).prop('onChangeComplete')({hex: '#123123'});
+    const sketchInput = screen.queryByDisplayValue('AABBCC');
+    await userEvent.type(sketchInput, 'cccccc');
```

Comment:
i think this should work using the `bindValueToOnChange` implemented in https://github.com/sulu/sulu/pull/6795 🙂